proto/Makefile

```diff
@@ -33,3 +33,9 @@ build: $(patsubst %.proto, %.pb.go, $(shell find src/determined -type f -name '*
 .PHONY: check
 check:
 	buf check lint
+
+.PHONY: fmt
+fmt:
+	type clang-format > /dev/null || exit 0
```

Comment:
IMO we should add this to CI if we're going to do this.

Reply:
hmm do you mean we should add the dependency? we don't really run fmt in CI. it'd be good to have `check` in CI which we don't have here 

I have this line here because I didn't want to worry about adding this dependency to our toolchain just yet and opportunistically format for now.

Reply:
Well, we've seen from the reST case how well it works to expect people to apply this sort of thing if they don't actually have to. :stuck_out_tongue: And there's a standard package for a reasonably recent version (8) even on Ubuntu 16.04, so it shouldn't be that big a deal to install. Moderate vote from me for going straight to checking it automatically, since we know the tool is stable.

(There's not a handy `--check`, but it looks like you can do `clang-format --dry-run --Werror` to get checking behavior.)

Reply:
I did add the check target with the same "don't fail if clang-format is missing". next would be adding the dependency to ci and contribution guide